include/lbann/utils/summary_impl.hpp

```diff
@@ -194,6 +195,9 @@ inline void lbann_summary::reduce_2norm(const std::string tag, const El::Abstrac
 
 template <typename TensorDataType>
 inline auto lbann_summary::local_sum(const El::AbstractMatrix<TensorDataType>& mat) const -> BiggerOf<TensorDataType, float> {
+#ifdef LBANN_HAS_CALIPER
+  CALI_CXX_MARK_FUNCTION;
+#endif
```

Comment:
I would say that we should define this pattern at a higher level:
```
// Maybe base.hpp, maybe elsewhere
#ifdef LBANN_HAS_CALIPER
#define LBANN_CALI_MARK_FUNC CALI_CXX_MARK_FUNCTION
#else
#define LBANN_CALI_MARK_FUNC
#endif
```
Then,
```suggestion
  LBANN_CALI_MARK_FUNC;
```